### PR TITLE
Varnish60

### LINKED ISSUE #137: ncsa_format_string needs "%" escaped for systemd
### Cookbook version
master

### Chef-client version
13.2.20

### Platform Details
Ubuntu 16

### Scenario:
Enable the logging service with the `varnishncsa` format.

### Steps to Reproduce:
Set `default['varnish']['configure']['ncsa']['action'] = :configure` and install on a systemd platform.

### Expected Result:
```
192.168.32.1|-|-|[09/Jul/2017:13:34:42 +0000]|"GET http://mysite.local/ HTTP/1.1"|200|52109|"-"|"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36 OPR/46.0.2597.32"
```

### Actual Result:
```
/|-|root|/run|"/system.slice"|/bin/sh|f46fb581373b43fd9ed9374f3e3091d3|"http://mysite.local/"|"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36 OPR/46.0.2597.32"
```

### PR DESCRIPTION
**Pull request by RyanJarv**
### Description

* Adds support Varnish 6.0
* Allow systemd platforms to set instance names. PR #157 and #153
  * Note: In some cases (instance_name is set when using systemd) this will change the varnish instance name.
* Escape ncsa_format_string for systemd. Fixes #137
* Don't set -n when instance_name is nil or empty on systemd

### Issues Resolved

* #137 
* #153 
* #157 

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by RyanJarv** [comment]
This bumps the major version to 4 as well

File: .kitchen.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ driver:
 provisioner:
   name: chef_zero
   require_chef_omnibus: 12.19.36
+  #require_chef_omnibus: 13.8.5
```
Comment (damacus) [inline .kitchen.yml]: You're installing using chef 12 everywhere?

We will be dropping 12 support soon

File: test/integration/varnish6/serverspec/default_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Encoding: utf-8
+
+require_relative 'spec_helper'
```
Comment (damacus) [inline test/integration/varnish6/serverspec/default_spec.rb]: Could you make these inspec instead of serverspec please?

File: .kitchen.yml (hunk 1 above)
Comment (RyanJarv) [reply .kitchen.yml]: Yeah so was planning to do that in another PR and this line got added here accidentally. But we can get that included here.

**Comment by damacus** [comment]
I'm going to resolve all of my comments in another PR. But this looks good thanks @ryanjarv

**Comment by damacus** [comment]
@RyanJarv you might want to check the CHANGELOG on master to correctness.

**Pull request merged**